src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php

```diff
@@ -58,10 +69,174 @@ public function getStatesAction(Request $request)
+                'prestashop.core.form.identifiable_object.handler.state_form_handler'
+            );
+
+            $stateForm = $stateFormBuilder->getFormFor((int) $stateId);
+            $stateForm->handleRequest($request);
+            $result = $stateFormHandler->handleFor((int) $stateId, $stateForm);
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_states_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/edit.html.twig', [
+            'stateForm' => null !== $stateForm ? $stateForm->createView() : null,
+            'stateInformation' => $stateInformation,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * @return array
+     */
+    private function getErrorMessages(): array
+    {
+        return [
+            StateException::class => $this->trans(
+                'Unexpected error occurred.',
+                'Admin.Notifications.Error'
+            ),
+            StateNotFoundException::class => $this->trans(
+                'The object cannot be loaded (or found)',
+                'Admin.Notifications.Error'
+            ),
+            StateConstraintException::class => [
+                StateConstraintException::INVALID_ID => $this->trans(
+                    'The object cannot be loaded (the identifier is missing or invalid)',
+                    'Admin.Notifications.Error'
+                ),
+                StateConstraintException::INVALID_FIELDS => $this->trans(
+                    'An error occurred when attempting to update the required fields.',
+                    'Admin.Notifications.Error'
+                ),
+                StateConstraintException::INVALID_NAME => $this->trans(
+                    'The %s field is invalid.',
+                    'Admin.Notifications.Error',
+                    [sprintf('"%s"', $this->trans('Name', 'Admin.Global'))]
+                ),
+                StateConstraintException::INVALID_ISO_CODE => $this->trans(
+                    'The %s field is invalid.',
+                    'Admin.Notifications.Error',
+                    [sprintf('"%s"', $this->trans('ISO code', 'Admin.International.Feature'))]
+                ),
+                StateConstraintException::INVALID_STATE => $this->trans(
+                    'An error occurred when attempting to update the required fields.',
+                    'Admin.Notifications.Error'
+                ),
+            ],
+            CannotUpdateStateException::class => $this->trans(
+                'An error occurred on saving.',
```

Comment:
I find this wording quite weird when reading it like this... what do you think about writing _An error occurred while attempting to save._ instead?